src/core/expression/qgsexpressionnodeimpl.cpp

```diff
@@ -532,39 +532,39 @@ QVariant QgsExpressionNodeBinaryOperator::evalNode( QgsExpression *parent, const
         bool matches;
         if ( mOp == boLike || mOp == boILike || mOp == boNotLike || mOp == boNotILike ) // change from LIKE syntax to regexp
         {
-          QString esc_regexp = QRegExp::escape( regexp );
+          QString esc_regexp = QRegularExpression::escape( regexp );
           // manage escape % and _
           if ( esc_regexp.startsWith( '%' ) )
           {
             esc_regexp.replace( 0, 1, QStringLiteral( ".*" ) );
           }
-          QRegExp rx( "[^\\\\](%)" );
+          QRegExp rx1( QStringLiteral( "[^\\\\](%)" ) );
           int pos = 0;
-          while ( ( pos = rx.indexIn( esc_regexp, pos ) ) != -1 )
+          while ( ( pos = rx1.indexIn( esc_regexp, pos ) ) != -1 )
           {
             esc_regexp.replace( pos + 1, 1, QStringLiteral( ".*" ) );
             pos += 1;
           }
-          rx.setPattern( QStringLiteral( "\\\\%" ) );
-          esc_regexp.replace( rx, QStringLiteral( "%" ) );
+          QRegExp rx2( QStringLiteral( "\\\\%" ) );
+          esc_regexp.replace( rx2, QStringLiteral( "%" ) );
           if ( esc_regexp.startsWith( '_' ) )
           {
             esc_regexp.replace( 0, 1, QStringLiteral( "." ) );
           }
-          rx.setPattern( QStringLiteral( "[^\\\\](_)" ) );
+          QRegExp rx3( QStringLiteral( "[^\\\\](_)" ) );
           pos = 0;
-          while ( ( pos = rx.indexIn( esc_regexp, pos ) ) != -1 )
+          while ( ( pos = rx3.indexIn( esc_regexp, pos ) ) != -1 )
           {
             esc_regexp.replace( pos + 1, 1, '.' );
             pos += 1;
           }
-          rx.setPattern( QStringLiteral( "\\\\_" ) );
-          esc_regexp.replace( rx, QStringLiteral( "_" ) );
+          const thread_local QRegularExpression rx4( QStringLiteral( "\\\\_" ) );
```

Comment:
Does this even need to be a regular expression? A standard replace would likely be much faster